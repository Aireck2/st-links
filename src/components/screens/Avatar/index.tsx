import { useGetGithub } from "@/hooks/useGetGithub";

import { AvatarContainerStyled, AvatarStyled, NotImage } from "./styles";

const Avatar = () => {
  const { data } = useGetGithub();
  return (
    <AvatarContainerStyled>
      {data?.avatarUrl ? (
        <AvatarStyled
          priority
          src={data.avatarUrl}
          alt="Profile picture"
          height={120}
          width={120}
        />
      ) : (
        <NotImage />
      )}
    </AvatarContainerStyled>
  );
};
export default Avatar;
