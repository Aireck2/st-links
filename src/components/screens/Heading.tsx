"use client";

import { FC } from "react";
import Image from "next/image";
import styled, { useTheme } from "styled-components";

import { constants } from "@/utils/constants";

const HeadingStyled = styled.section`
  display: flex;
  align-items: center;
  grid-gap: 2rem;
  flex-direction: column;
  .brand {
    font-weight: bold;
    color: ${({ theme }) => theme.colors.light};
  }
  .social {
    display: flex;
    grid-gap: 1rem;
    min-width: 200px;
    justify-content: space-between;
    svg {
      color: white;
    }
  }
`;

const AvatarStyled = styled.div`
  width: 120px;
  height: 120px;
  background-color: ${({ theme }) => theme.colors.light};
  border-radius: 2rem;
`;

export const Heading: FC = () => {
  const theme = useTheme();
  return (
    <HeadingStyled>
      <AvatarStyled />
      <p className="brand">@erickescribaa</p>
      <div className="social">
        {constants.social.map((sm) => (
          <a href={sm.url} key={sm.name}>
            <Image
              key={sm.name}
              src={sm.src}
              alt={sm.name}
              height={24}
              width={24}
            />
          </a>
        ))}
      </div>
    </HeadingStyled>
  );
};
