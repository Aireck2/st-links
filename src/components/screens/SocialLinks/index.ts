export { default as HorizontalSocialLinks } from "./HorizontalSocialLinks";
export { default as VerticalSocialLinks } from "./VerticalSocialLinks";
