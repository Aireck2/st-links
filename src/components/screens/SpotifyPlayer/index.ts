export { default as NotPlaying } from "./NotPlaying";
export { default as SpotifyPlayer } from "./SpotifyPlayer";
export { default as PlayingProgress } from "./PlayingProgress";
