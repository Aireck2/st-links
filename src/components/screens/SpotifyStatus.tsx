"use client";

import { FC } from "react";
import Image from "next/image";

import styled from "styled-components";

import { AdaptSpotifyData } from "@/types/api/spotify";

const common = `
  border-radius: 1rem;
`;
const br = "1rem";
const bgBr = "2rem";

const HeadingStyled = styled.section`
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: ${bgBr};
  margin: auto;
  max-width: 700px;
  box-shadow: 4px 4px 12px #121212, -4px -4px 12px #313131;
  .albumImage {
    ${common};
    @media screen and (min-width: 576px) {
      width: 100%;
      height: 100%;
      max-height: 150px;
      max-width: 150px;
    }
  }
  position: relative;
  .bg-image {
    position: absolute;
    z-index: 10;
    object-fit: cover;
    border-radius: ${bgBr};
  }
  .container {
    width: 100%;
    height: 100%;
    backdrop-filter: blur(1rem);
    background-color: rgba(0, 0, 0, 0.7);
    border-radius: ${bgBr};
    position: relative;
    z-index: 20;
    object-fit: cover;
    padding: 2rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    grid-gap: 20px;
    @media screen and (min-width: 480px) {
      flex-direction: row;
    }
    .content {
      min-width: 100%;
      align-self: flex-start;
      @media screen and (min-width: 480px) {
        min-width: 260px;
      }
    }
    .album-title {
      color: #ccc;
      font-size: 16px;
      font-weight: normal;
      font-size: clamp(0.8rem, 2vw, 1rem);
    }
    .song-title {
      color: white;
      font-size: clamp(0.9rem, 3.5vw, 1.4rem);
      font-weight: normal;
    }
    .artist-title {
      color: #ccc;
      font-weight: normal;
      font-size: clamp(0.8rem, 2vw, 1rem);
    }
  }
`;
const NotPlayingStyled = styled.section`
  display: flex;
  justify-content: center;
  .icon {
    display: grid;
    place-items: center;
    padding: 1rem;
    background-color: #212121;
    border-top-left-radius: 2rem;
    border-bottom-left-radius: 2rem;
    @media screen and (min-width: 480px) {
      padding: 2rem;
    }
  }
  .content {
    padding: 1rem;
    background-color: #121212;
    border-top-right-radius: 2rem;
    border-bottom-right-radius: 2rem;
    @media screen and (min-width: 480px) {
      padding: 2rem;
    }

    p {
      color: #b3b3b3;
    }
  }
`;

interface SpotifyStatusProps {
  data: AdaptSpotifyData;
}

export const SpotifyStatus: FC<SpotifyStatusProps> = ({ data }) => {
  const caption = "I'm coding & vibing with this song!";

  return data.isPlaying ? (
    <HeadingStyled>
      <Image
        className="bg-image"
        priority
        src={data.albumImageUrl}
        alt={`props.album`}
        fill
      />
      <div className="container">
        <Image
          className="albumImage"
          priority
          src={data.albumImageUrl}
          alt={`props.album`}
          height={100}
          width={100}
        />
        <div className="content">
          <h3 className="album-title">{caption}</h3>
          <h2 className="song-title">{data.title}</h2>
          <h4 className="artist-title">
            {data.artists} â€” {data.album}
          </h4>
        </div>
      </div>
    </HeadingStyled>
  ) : (
    <NotPlayingStyled>
      <div className="icon">
        <Image
          src="/icons/spotify.svg"
          alt={"Spotify icon"}
          width={50}
          height={50}
        ></Image>
      </div>
      <div className="content">
        <p>Not playing right now...</p>
        <p>I bet I&#39;m studying or sleeping</p>
      </div>
    </NotPlayingStyled>
  );
};
