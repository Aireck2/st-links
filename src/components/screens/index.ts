"use client";
export * from "./Heading/Heading";
export * from "./SpotifyPlayer";
export * from "./SocialLinks";
