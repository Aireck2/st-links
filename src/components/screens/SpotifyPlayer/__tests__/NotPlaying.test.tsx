/**
 * @jest-environment jsdom
 */

import { NotPlaying } from "@/components/screens";
import { render } from "@testing-library/react";
import "jest-styled-components";

describe("NotPlaying", () => {
  test("render text", () => {
    const textIn = "Not playing right now...";

    const component = render(<NotPlaying />);
    const text = component.getByText(textIn);

    expect(text.textContent).toBe(textIn);
  });

  test("display svg image", () => {
    render(<NotPlaying />);
    const image = document.querySelector("img");

    expect(image?.src).toContain("spotify.svg");
  });

  test("display correct size of svg image", () => {
    render(<NotPlaying />);
    const image = document.querySelector("img");

    expect(image?.width).toBe(50);
    expect(image?.height).toBe(50);
  });

  test("display correct styles", () => {
    const component = render(<NotPlaying />);

    expect(component.container.firstChild).toMatchSnapshot();
  });
});
