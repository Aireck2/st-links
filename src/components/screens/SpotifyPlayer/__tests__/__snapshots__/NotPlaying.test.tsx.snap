// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotPlaying display correct styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .icon {
  display: grid;
  place-items: center;
  padding: 1rem;
  background-color: #212121;
  border-top-left-radius: 1rem;
  border-bottom-left-radius: 1rem;
}

.c0 .content {
  padding: 1rem;
  background-color: #121212;
  border-top-right-radius: 1rem;
  border-bottom-right-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .content p {
  color: #b3b3b3;
  font-size: clamp(0.8rem,3vw,1rem);
}

<section
  class="c0"
>
  <div
    class="icon"
  >
    <img
      alt="Spotify icon"
      data-nimg="1"
      decoding="async"
      height="50"
      loading="lazy"
      src="/icons/spotify.svg"
      style="color: transparent;"
      width="50"
    />
  </div>
  <div
    class="content"
  >
    <p>
      Not playing right now...
    </p>
    <p>
      I bet I'm studying or sleeping
    </p>
  </div>
</section>
`;
